cohere_aws/client.py

```diff
@@ -387,13 +387,15 @@ def rerank(self,
                documents: Union[List[str], List[Dict[str, Any]]],
                top_n: Optional[int] = None,
                variant: Optional[str] = None,
-               max_chunks_per_doc: Optional[int] = None) -> Reranking:
+               max_chunks_per_doc: Optional[int] = None,
+               rank_fields: Optional[List[str]] = None) -> Reranking:
         """Returns an ordered list of documents oridered by their relevance to the provided query
         Args:
             query (str): The search query
             documents (list[str], list[dict]): The documents to rerank
             top_n (int): (optional) The number of results to return, defaults to return all results
             max_chunks_per_doc (int): (optional) The maximum number of chunks derived from a document
+            rank_fields (list[str]): (optional) The fields used for reranking
```

Comment:
is it compatible with all versions or just v3?

Reply:
we should block just for v3

Reply:
Do we know the version from the sdk?

Reply:
I don't think we can enforce the constraint in code, but at least the docstring can say that it's not supported pre-v3

Reply:
@leilacc im guessing there is really no way to know from the ARN or any meta data around what model is being used (like backbone whether its v2/v3). 

I think it's fine to just warn + have in doc-string